aw_client/client.py

```diff
@@ -187,6 +196,7 @@ def _load_queue(self):
                     logger.warning("Skipping request that failed to load")
 
         # Insert failed_requests into dispatching queue
+        # FIXME: We really shouldn't be clearing the file here until the events have been sent to server.
```

Comment:
I know it's not a great solution, but to avoid event duplication in case of a crash this works.

We should probably have to reconsider the whole dispatchThreads behavior of saving to file, and try to only save to file when the client is shutting down and only load the queue from file at startup. It's harder to implement and unless the client is force closed for some reason it should be fool-proof.